addons/purchase/tests/test_purchase.py

```diff
@@ -483,6 +483,46 @@ def test_purchase_multicurrency(self):
 
         self.assertEqual(order_b.order_line.price_unit, 10.0, 'The price unit should be 10.0')
 
+    def test_discount_update_when_no_seller_found(self):
+        """ Purchase order line should reset discount when no vendor found based on quantity
+        """
+        vendor = self.env['res.partner'].create({
+            'name': 'Test Vendor',
+            'supplier_rank': 1,
+        })
+
+        # Create product with one vendor and discount based on quantity
+        product = self.env['product.product'].create({
+            'name': 'Product',
+            'seller_ids': [
+                (0, 0, {
+                    'partner_id': vendor.id,
+                    'min_qty': 10,
+                    'price': 10,
+                    'discount': 10
+                }),
+                (0, 0, {
+                    'partner_id': vendor.id,
+                    'min_qty': 20,
+                    'price': 10,
+                    'discount': 15
+                })
+            ]
+        })
+
+        # Create new PO line with the product
+        po_form = Form(self.env['purchase.order'])
+        po_line = po_form.order_line.new()
+        po_line.product_id = product
+
+        # The discount should update accordingly based on quantity
+        po_line.product_qty = 10
+        self.assertEqual(po_line.discount, 10, "first seller should be selected so discount should be 10")
+        po_line.product_qty = 22
+        self.assertEqual(po_line.discount, 15, "second seller should be selected so discount should be 15")
+        po_line.product_qty = 1
+        self.assertEqual(po_line.discount, 0, "no seller should be selected so discount should be 0")
```

Comment:
Can you also check the price ?
So we ensure that both discount and value are ok ?
(To do so, it's worth adding a default price on your product)